### ISSUE TITLE
Some blocks are lost.

### ISSUE THREAD

**Issue by bigjoker007** [node-real/bsc-erigon#395]
I used the v1.2.5 and the 20240416 snapshot to start the node，now it has synchronized to the block #38696368. 
But the blocks between #36500000 and #37806999 are lost, all APIs return None.

The flag:

> --log.dir.path /hdd/app/BscNode/logs
> --datadir /ssd/data/erigon 
> --chain=bsc 
> --http 
> --http.port=6666 
> --http.addr 0.0.0.0 
> --http.api=web3,net,eth,debug,trace,txpool 
> --maxpeers=5000 
> --db.pagesize=16k

And the hard drive still has 1.3TB of available space.

**Comment by zlacfzy** [comment]
Use the latest release.  
```
make integration
./integration stage_headers --reset --chain=bsc --datadir=./data 
```

**Comment by wtdcode** [comment]
@blxdyx I'm having the same issue with two bsc-erigon nodes.

Is it possible that bsc-erigon wrongly determine the max block in snapshots? As https://github.com/node-real/bsc-erigon-snapshot suggests, the snapshots should be way beyond 36500000.

**Comment by wtdcode** [comment]
> Yes, header reset will fix

No, not exactly. For my local setup, my local snapshots files indicates more than 36500000 blocks but bsc-erigon just recognizes the 36500000 blocks. For instance, I have:

```
-rw-r--r--  1 erigon erigon   86M Jun  9 17:32 v1-039000-039100-transactions.idx
-rw-r--r--  1 erigon erigon  4.2G Jun 10 22:53 v1-039000-039100-transactions.seg
-rw-r--r--  1 erigon erigon   42K Jun 10 22:50 v1-039000-039100-transactions.seg.torrent
-rw-r--r--  1 erigon erigon   48M Jun  9 17:32 v1-039000-039100-transactions-to-block.idx
```

but erigon seems ignoring these snapshots and thinks the highest block in snapshots is 36500000.

Therefore, when erigon syncs another 500000 blocks, the blocks with will be lost *AGAIN*.

**Comment by wtdcode** [comment]
> 1. use the latest branch, it update the snapshots between [0, 38700000)
> 2. which range block your have lost.

Yes, I noticed you have updated the hashes to https://github.com/node-real/bsc-erigon-snapshot and I have tried both master branch and v1.2.9.

I'm losing the blocks after 36000000 although I have all snapshots according to the latest hashes in the repo.

**Comment by wtdcode** [comment]
@blxdyx Specially, I'm repeatlly suffering from:

```
erigon-1  | [WARN] [06-11|10:54:35.752] [1/15 Snapshots] Some blocks are not in snapshots and not in db max_in_snapshots=35999999 min_in_db=36874500
```

even though I have snapshots containing blocks after 35999999

**Comment by wtdcode** [comment]
One thing while I'm debugging code is that, the preverified hashes are empty here:

https://github.com/node-real/bsc-erigon/blob/30b23c9d75dabe486d2f87cdfea3ae3af436a2ad/turbo/snapshotsync/snapshotsync.go#L102

I can debug a bit further later.

**Comment by wtdcode** [comment]
> Reset headers don't work?

I works for the first a few days. After syncing another a few blocks (more than 0.5M so that erigon will produce snapshots), the blocks within will be lost again.

**Comment by wtdcode** [comment]
From my understanding of erigon code base, the issue should be fixed by #405 but I will continue monitoring my node status and report in case something is wrong.

For other guys getting this error, simply apply #405 and remove `prohibit_new_downloads.lock` under the `snapshots/` directory.
